deployment/helm/templates/custom-discovery.yaml

```diff
@@ -0,0 +1,98 @@
+  brokerPodSpec:
+    containers:
+    - name: {{ .Values.customDiscovery.name }}-broker
+      image: {{ .Values.customDiscovery.brokerPod.image | quote }}
+      {{- with .Values.customDiscovery.pullPolicy }}
+      imagePullPolicy: {{ . }}
+      {{- end }}
+      resources:
+        limits:
+          {{`"{{PLACEHOLDER}}"`}} : "1"
+    {{- with .Values.imagePullSecrets }}
+    imagePullSecrets:
+      {{- toYaml . | nindent 6 }}
+    {{- end }}
+  {{- end }}
+  {{- if .Values.customDiscovery.brokerPod.image }}
+  {{- if .Values.customDiscovery.createInstanceServices }}
+  instanceServiceSpec:
+    type: {{ .Values.customDiscovery.instanceService.type }}
+    ports:
+    - name: {{ .Values.customDiscovery.instanceService.name }}
+      port: {{ .Values.customDiscovery.instanceService.port }}
+      protocol: {{ .Values.customDiscovery.instanceService.protocol }}
+      targetPort: {{ .Values.customDiscovery.instanceService.targetPort }}
+  {{- end }}
+  {{- if .Values.customDiscovery.createConfigurationService }}
+  configurationServiceSpec:
+    type: {{ .Values.customDiscovery.configurationService.type }}
+    ports:
+    - name: {{ .Values.customDiscovery.configurationService.name }}
+      port: {{ .Values.customDiscovery.configurationService.port }}
+      protocol: {{ .Values.customDiscovery.configurationService.protocol }}
+      targetPort: {{ .Values.customDiscovery.configurationService.port }}
+  {{- end }}
+  {{- end }}
+  {{- if .Values.customDiscovery.properties }}
+  properties:
+  {{- toYaml . | nindent 4 }}
+  {{- else }}
+  properties: {}
+  {{- end }}
+  capacity: {{ .Values.customDiscovery.capacity }}
+{{- end }}
+---
+{{- if .Values.customDiscovery.discovery.enabled }}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ printf "%s-discovery-daemonset" .Values.customDiscovery.name | quote }}
+spec:
+  selector:
+    matchLabels:
+      name: {{ printf "%s-discovery" .Values.customDiscovery.name | quote }}
+  template:
+    metadata:
+      labels:
+        name: {{ printf "%s-discovery" .Values.customDiscovery.name | quote }}
+    spec:
+      containers:
+      - name: {{ printf "%s-discovery" .Values.customDiscovery.name | quote }}
+        image: {{ required "A customDiscovery.discovery.image.repository is required." .Values.customDiscovery.discovery.image.repository | quote }}
+        {{- with .Values.customDiscovery.discovery.image.pullPolicy }}
+        imagePullPolicy: {{ . }}
+        {{- end}}
+        volumeMounts:
+        - name: agent-registration
+          mountPath: /var/lib/akri
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- if .Values.customDiscovery.discovery.nodeSelectors }}
+      nodeSelector:
+      {{- toYaml .Values.customDiscovery.discovery.nodeSelectors | nindent 8 }}
+      {{- end }}
+      volumes:
+      - name: agent-registration
+        hostPath:
+          path: /var/lib/akri
```

Comment:
i wonder if hostPath shouldn't be configurable with /var/lib/akri as a default.

Reply:
I see it as always needing to be mounted since every Discovery Handler needs to register with the Agent's registration socket

Reply:
something always needs to be mounted for sure.  but i think in the past it has proven handy to make the value configurable ... something like this in agent and in all of the DH pods:

```
      - name: agent-registration
        hostPath:
          path: "{{ .Values.agent.registrationSocketPath }}"
```

Reply:
fixed in 00cc473d7bbd8e3